services/network/logicalnetwork/logicalnetwork.go

```diff
@@ -0,0 +1,271 @@
+		return nil, errors.Wrapf(errors.InvalidInput, "Location is not specified")
+	}
+	wssdnetwork := &wssdcloudnetwork.LogicalNetwork{
+		Name:         *c.Name,
+		LocationName: *c.Location,
+		Tags:         tags.MapToProto(c.Tags),
+	}
+
+	if c.Version != nil {
+		if wssdnetwork.Status == nil {
+			wssdnetwork.Status = status.InitStatus()
+		}
+		wssdnetwork.Status.Version.Number = *c.Version
+	}
+
+	if c.LogicalNetworkPropertiesFormat != nil {
+		subnets, err := getWssdNetworkSubnets(c.Subnets)
+		if err != nil {
+			return nil, err
+		}
+		wssdnetwork.Subnets = subnets
+
+		if c.LogicalNetworkPropertiesFormat.MacPoolName != nil {
+			wssdnetwork.MacPoolName = *c.LogicalNetworkPropertiesFormat.MacPoolName
+		}
+	}
+
+	return wssdnetwork, nil
+}
+
+func ipAllocationMethodProtobufToSdk(allocation wssdcommonproto.IPAllocationMethod) network.IPAllocationMethod {
+	switch allocation {
+	case wssdcommonproto.IPAllocationMethod_Static:
+		return network.Static
+	case wssdcommonproto.IPAllocationMethod_Dynamic:
+		return network.Dynamic
+	}
+	return network.Dynamic
+}
+
+func ipAllocationMethodSdkToProtobuf(allocation network.IPAllocationMethod) wssdcommonproto.IPAllocationMethod {
+	switch allocation {
+	case network.Static:
+		return wssdcommonproto.IPAllocationMethod_Static
+	case network.Dynamic:
+		return wssdcommonproto.IPAllocationMethod_Dynamic
+	}
+	return wssdcommonproto.IPAllocationMethod_Dynamic
+}
+
+func getWssdNetworkIPPoolInfo(ippoolinfo *network.IPPoolInfo) *wssdcommonproto.IPPoolInfo {
+	if ippoolinfo != nil {
+		return &wssdcommonproto.IPPoolInfo{
+			Used:      ippoolinfo.Used,
+			Available: ippoolinfo.Available,
+		}
+	}
+	return nil
+}
+func getWssdNetworkSubnets(subnets *[]network.LogicalSubnet) (wssdsubnets []*wssdcloudnetwork.LogicalSubnet, err error) {
+	if subnets == nil {
+		return
+	}
+
+	for _, subnet := range *subnets {
+		wssdsubnet := &wssdcloudnetwork.LogicalSubnet{}
+		if subnet.Name == nil {
+			err = errors.Wrapf(errors.InvalidInput, "Subnet name is missing")
+			return
+		}
+		wssdsubnet.Name = *subnet.Name
+
+		if subnet.LogicalSubnetPropertiesFormat == nil {
+			continue
+		}
+
+		if subnet.Vlan == nil {
+			wssdsubnet.Vlan = 0
+		} else {
+			wssdsubnet.Vlan = uint32(*subnet.Vlan)
+		}
+
+		wssdsubnetRoutes, err1 := getWssdNetworkRoutes(subnet.RouteTable)
+		if err1 != nil {
+			err = err1
+			return
+		}
+		wssdsubnet.Routes = wssdsubnetRoutes
+		wssdsubnet.Allocation = ipAllocationMethodSdkToProtobuf(subnet.IPAllocationMethod)
+
+		if subnet.AddressPrefix != nil {
+			wssdsubnet.AddressPrefix = *subnet.AddressPrefix
+		}
+
+		// An address prefix is required if using ippools
+		if len(subnet.IPPools) > 0 && subnet.AddressPrefix == nil {
+			err = errors.Wrapf(errors.InvalidInput, "AddressPrefix is missing")
+			return
+		}
+
+		if subnet.DhcpOptions != nil && subnet.DhcpOptions.DNSServers != nil {
+			wssdsubnet.Dns = &wssdcommonproto.Dns{
+				Servers: *subnet.DhcpOptions.DNSServers,
+			}
+		}
```

Comment:
why did we decide to use dhcp options for moc-sdk but just dnsservers for cloudagent?